application.cpp

```diff
@@ -445,14 +462,18 @@ void application::exec() {
       (void)work;
       bool more = true;
       while( more || io_serv->run_one() ) {
+         if (my->_is_quiting)
+            break;
          while( io_serv->poll_one() ) {}
          // execute the highest priority item
          more = pri_queue.execute_highest();
       }
+      pri_queue.clear(); // make sure the queue is empty
```

Comment:
Why does this need to be done? In just a few lines later we destroy `application` anyways. Is it important to have it clear in `shutdown()`?

Reply:
I don't have a strong evidence that it is necessary, however it may be possible that some of the handlers posted on the priority queue have references to the plugins, and may access these references in handler destructors.
Since we destroy the plugins in `application::shutdown()`, it seems to me like a good precaution to mage sure that any of the queued handlers are destroyed before. And there is no downside of doing so that I can see.